src/tree/registries/RegistriesTreeItem.ts

```diff
@@ -132,8 +132,10 @@ export class RegistriesTreeItem extends AzExtParentTreeItem {
             cachedProvider.url = wizardContext.url;
             cachedProvider.username = wizardContext.username;
 
-            if (wizardContext.password) {
-                await setRegistryPassword(cachedProvider, wizardContext.password);
+            if (wizardContext.secret && provider.authHelper) {
+                await provider.authHelper.persistAuth(cachedProvider, wizardContext.secret);
+            } else if (wizardContext.secret) {
+                await setRegistryPassword(cachedProvider, wizardContext.secret);
```

Comment:
Would it make sense to push this into an `authHelper` on the other registry providers, so that the parent type doesn't have to make this distinction?